modules/promotions/admin-menu-items/base-promotion-item.php

```diff
@@ -33,17 +33,31 @@ public function get_image_url() {
 	}
 
 	public function render() {
+		$config = [
+			'title' => $this->get_promotion_title(),
+			'description' => $this->get_promotion_description(),
+			'image' => $this->get_image_url(),
+			'upgrade_text' => $this->get_cta_text(),
+			'upgrade_url' => $this->get_cta_url(),
+		];
+
+		$config = apply_filters( 'elementor/' . $this->get_name() . '/restrictions/custom_promotion', $config );
```

Comment:
```suggestion
		$config = apply_filters( 'elementor/' . $this->get_name() . '/custom_promotion', $config );
```
I don't think `restrictions` make sense for all the promotions?

Reply:
> Do you need to create a get_name() in this base class?
Yes because you insisted on me creating `get_name()` @louiswol94. This PR started out perfectly functional and the refactor we made at your request in this file might be beneficial to future extension of related promotions. But it only affected one promotion in this PR. So it's inflated this PR quite a lot because this ended up dragging in unrelated files into the PR. But now we're just going in circles. 

> I don't think restrictions make sense for all the promotions?
Naming flows quite nicely at the moment except for the admin menu change requested. As you can see not all filters had 'restrction' only role manager to follow the naming of the action and fonts which is related. 
<img width="309" alt="image" src="https://github.com/elementor/elementor/assets/130742176/31b4a5a6-5ad9-4e62-8f31-fcdfd8742b9e">

I'm shortening both font and role manager naming. 
<img width="309" alt="image" src="https://github.com/elementor/elementor/assets/130742176/c06dd9ed-7c98-4e79-b98f-222dd32e4540">